src/Main.ts

```diff
@@ -1114,6 +1114,11 @@ export class Main {
             }
             room.setBotClient(slackClient);
             room.SlackChannelName = infoRes.channel.name;
+
+            // Perform syncing asyncronously.
+            this.teamSyncer?.syncMembershipForRoom(matrixRoomId, opts.slack_channel_id, opts.team_id, slackClient).catch((ex) => {
```

Comment:
Do you not want to await the sync, so it doesn't race with the code below?

Reply:
Moved this to the end.